### DISCUSSION TITLE
RTX 4090 better than L40 ?

### DISCUSSION THREAD

**Discussion by alessandroassirelli98** [isaac-sim/IsaacLab#1310] [Q&A] [answered]
Hello!
I was checking out the performance benchmarks on various environments using RTX 4090 and L40. It is weird to me that the single-node performances are inferior on the L40 GPU compared to the 4090. Shouldn't it be the opposite? L40 has more Cuda Cores, more RT cores, and more Tensor cores from the same generation. 
Also by checking the performance on the single node, multi GPU I see that when using 4 GPUs the performance increase by a factor of 4.03. Isn't it too much?
- FPS on single node single 4090 GPU G1 training: 94000
- FPS on single node single L40 GPU G1 training: 72000
- FPS on single node four L40 GPUs G1 training: 290000

from here https://images.nvidia.com/content/Solutions/data-center/vgpu-L40-datasheet.pdf I see that it doesn't even support NVLink, so the communication might be a bottleneck. Is it possible by chance the data for RTX4090 and L40 on the single node single GPU are swapped😅? That to me would make more sense. Otherwise piece of cake, a cheaper gpu outperform the datacenter one🤙 !

Also I was wondering what are the specs that matter the most for RL training. Since the training happens in headless mode, are RT cores that important?
If you have any advice on choosing the best setup for RL training please share!

Thanks!

**Comment by glvov-bdai** [comment, marked as answer]
Hi I believe these results are correct. I believe that while the 4090 is more performant than the L40, it also has less VRAM (48 gb vram L40 vs 24 gb vram 4090). This means that the L40 theoretically speaking could simulate much more environments in parallel, and can handle workloads with larger tensors.

External L40 vs 4090 benchmark
https://technical.city/en/video/GeForce-RTX-4090-vs-L4
